tracing-subscriber/src/fmt/time/time_crate.rs

```diff
@@ -45,6 +47,23 @@ pub struct UtcTime<F> {
     format: F,
 }
 
+/// Formats the current time using a fixed offset and a [formatter] from the [`time` crate].
+///
+/// This is typically used as an alternative to [`LocalTime`]. `LocalTime` determines the offset
+/// every time it formats a message, which may be unsound or fail. With `OffsetTime`, the offset is
+/// determined once. This makes it possible to do so while the program is still single-threaded and
+/// handle any errors. However, this also means the offset cannot change while the program is
+/// running (the offset will not change across DST changes).
+///
+/// [formatter]: https://docs.rs/time/0.3/time/formatting/trait.Formattable.html
+/// [`time` crate]: https://docs.rs/time/0.3/time/
+#[derive(Clone, Debug)]
+#[cfg_attr(docsrs, doc(cfg(feature = "time")))]
```

Comment:
this states that the type requires the `time` feature, but it looks like all the _methods_ implemented for this type also require the `local-time` feature...should the type itself also require that feature flag?

Reply:
`OffsetTime::local_rfc_3339` requires `local-time`, but `OffsetTime::init` and `OffsetTime::format_time` (our `impl FormatTime`) do not. It should be possible (although admittedly unusual) to use `OffsetTime` with a non-local offset (not needing `time`'s `local-offset` feature at all). And if our caller uses `OffsetTime::init` and calls `current_local_offset` themselves it seemed reasonable to have them enable the `local-offset` feature on the `time` crate themselves, rather than having them use our `local-time` feature just so we can turn on that `time` feature for them (we won't be doing anything with local time ourselves).

Please let me know if I'm overlooking something.